drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -24,18 +24,25 @@ LOG_MODULE_REGISTER(esp32_wifi, CONFIG_WIFI_LOG_LEVEL);
 #include <esp_mac.h>
 #include "wifi/wifi_event.h"
 
+K_MUTEX_DEFINE(lock);
```

Comment:
a prefix that explains why the lock is necessary or what it is trying to protect?

Reply:
Removed this. Because of no need.